pkgs/development/python-modules/pydantic-core/default.nix

```diff
@@ -0,0 +1,79 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cargo
+, rustPlatform
+, rustc
+, typing-extensions
+, pythonRelaxDepsHook
+, pytestCheckHook
+, hypothesis
+, pytest-benchmark
+, pytest-timeout
+, pytest-mock
+, dirty-equals
+}:
+
+buildPythonPackage rec {
+  pname = "pydantic-core";
+  version = "2.1.3";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "pydantic";
+    repo = "pydantic-core";
+    rev = "v${version}";
+    hash = "sha256-JdwzCD7ppxfqf/KPMDtCyw+XaZOzyKKgd7AhtF7gVKg=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  nativeBuildInputs = [
+    cargo
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+    rustc
+    typing-extensions
+    pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
+
+  pythonRelaxDeps = [
+    # NOTE: bumping typing-extensions to 4.7.1 would
+    # cause 5000+ rebuilds at the time of submission
+    #
+    # TODO: remove when typing-extensions is > 4.7.0
+    "typing-extensions"
+  ];
+
+  pythonImportsCheck = [ "pydantic_core" ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    hypothesis
+    pytest-benchmark
```

Comment:
```suggestion
```
We don't want to use compute power for benchmarking

Reply:
It is disabled in `disabledTestPaths`, but still needs to be passed due to [`pyproject.toml` flags](https://github.com/pydantic/pydantic-core/blob/9be711e35760c248094367e0aa117ea7b3013909/pyproject.toml#L76-L81) that are only available via this plugin.

Reply:
Just patch them out to not trigger unnecessary rebuilds

Reply:
Oh yeah, good point! Thx!

Reply:
Done so in [force-push](https://github.com/NixOS/nixpkgs/compare/e6488e846cee733f352ed57c2415b9e5a1bf25db..173deded1e38c279baabe88a448e657ffc595edd). Also removed the typing-extensions relaxation after [the update has been merged into staging](https://github.com/NixOS/nixpkgs/pull/243861), already.